apis/datadoghq/v2alpha1/datadogagent_default.go

```diff
@@ -50,7 +50,7 @@ const (
 	defaultDogstatsdHostPortEnabled        bool   = false
 	defaultDogstatsdPort                   int32  = 8125
 	defaultDogstatsdSocketEnabled          bool   = true
-	defaultDogstatsdSocketPath             string = "/var/run/datadog/dsd.socket"
+	defaultDogstatsdSocketPath             string = "/var/run/datadog/statsd/dsd.socket"
```

Comment:
Changed path to match [DogstatsdSocketVolumePath](https://github.com/DataDog/datadog-operator/blob/3c5f80ad6710ece6800b6287c133668d19b0296b/apis/datadoghq/common/const.go#L191) that's mounted in the [agent defaulting code](https://github.com/DataDog/datadog-operator/blob/3c5f80ad6710ece6800b6287c133668d19b0296b/controllers/datadogagent/component/agent/default.go#L294)